### DISCUSSION TITLE
MS SQL Server Stored Procedure

### DISCUSSION THREAD

**Discussion by IvanGhisetti** [dbeaver/dbeaver#35142] [Q&A] [answered]
Hi all, i'm just starting using DBaver.
I connect it to my MS SQL DB but i have a problem using stored procedure with parameters. When i execute the procedure DBaver don't ask me for parameters of the SP like MSSMS does. Someone have a solution?
Thanks.

**Comment by TB-HunBoka** [comment, marked as answer]
Might not what you want, but you can write parameter placeholders in the query if you enable variables.
![image](https://github.com/user-attachments/assets/b71aa420-c738-428d-ad3e-4163a1203a18)
Then you can use the Named parameter prefix as:
`select :something from dual`

Then DBeaver will ask you about the parameter when you execute the query.

**Reply by IvanGhisetti** [reply]
Thanks a lot! now it's better but i have another question. I setup parameters like that
![image](https://github.com/user-attachments/assets/27797016-a349-40f3-a150-fba841eab5ef)
Now i'm able to call stored procedure and DBaver ask me for parameters
![image](https://github.com/user-attachments/assets/4b6ad06c-aa06-4268-938e-060c157bfa5c)
My last question: is it possible that when i type the prefix '@' to have the list of parameters inserted in my stored procedure? otherwise i need every time to check them before in the stored procedure definition and write them manually.....
